packages/github-actions/teams-toolkit-cicd/src/buildMapQuerier.ts

```diff
@@ -0,0 +1,47 @@
+import {Commands} from './constant'
+import {InternalError} from './errors'
+import {buildMap} from './buildMap'
+
+export class BuildMapQuerier {
+  static instance: BuildMapQuerier
+  static validOutputs: string[] = [Commands.NpmInstall, Commands.NpmRunBuild]
+
+  private constructor() {}
+  static async getInstance(): Promise<BuildMapQuerier> {
+    if (!BuildMapQuerier.instance) {
+      BuildMapQuerier.instance = new BuildMapQuerier()
+    }
+
+    return BuildMapQuerier.instance
+  }
+
+  query(cap: string, lang?: string): string[] {
+    const commands = this._query(cap, lang)
+
+    if (commands.some(value => !BuildMapQuerier.validOutputs.includes(value))) {
+      throw new InternalError('Invalid command/s found in buildMap.json')
```

Comment:
Nit: as you told me, it is not .json but .ts

Reply:
Legacy, update it later. Thanks for pointing it out since I almost forget about this in log/message.